src/evo/deterministicmns.cpp

```diff
@@ -960,10 +960,7 @@ bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, gsl::no
 
     // The payee for the current block was determined by the previous block's list, but it might have disappeared in the
     // current block. We still pay that MN one last time, however.
-    if (payee && newList.HasMN(payee->proTxHash)) {
-        auto dmn = newList.GetMN(payee->proTxHash);
-        // HasMN has reported that GetMN should succeed, enforce that.
-        assert(dmn);
+    if (auto dmn = payee ? newList.GetMN(payee->proTxHash) : payee) {
```

Comment:
we should never go inside that `if` for `payee == nullptr` or `dmn == nullptr`, so it should probably be more like
```suggestion
    if (auto dmn = newList.GetMN(payee ? payee->proTxHash : uint256())) {
```

Reply:
nvm 🙈

Reply:
note to myself: go get a cup of coffee before reviewing PRs in the morning :D

Reply:
I mean; I do kinda agree with pre-coffee Udjin. Just in a different manner; I think it makes sense to explicitly say nullptr here. 

```suggestion
    if (auto dmn = payee ? newList.GetMN(payee->proTxHash) : nullptr) {
```